tests/api-diff/class_detection.t

```diff
@@ -0,0 +1,55 @@
+Here we generate a `.mli` file with the class declaration:
+
+  $ cat > ref_class.mli << EOF
+  > class ref_class : object
+  >   method get : int
+  >   method set : int -> unit
+  > end 
+  > EOF
+
+We generate .cmi file
+  $ ocamlc ref_class.mli
+
+Now, we run api-diff on the same cmi file as both arguments, there should be no difference
+  $ api-diff ref_class.cmi ref_class.cmi
+
+### Adding a new class:
+
+Generate a new .mli file with an additional class
+  $ cat > add_class.mli << EOF
+  > class ref_class : object
+  >    method get : int 
+  >    method set : int -> unit
+  > end
+  > class add_class : object
+  >    method calculate : float -> float
+  > end
+  > EOF
+
+Compile the new .mli file to a .cmi file
+  $ ocamlc add_class.mli
+
+Run api-diff and check the output
+  $ api-diff ref_class.cmi add_class.cmi
+  diff module Add_class:
+  +type add_class = < calculate : float -> float >
+  +class add_class: class add_class : object method calculate : float -> float end
```

Comment:
As you can see here, there's an error in the code that pretty prints the class, it prints the `class <class_name>` header twice!

Reply:
This is still not resolved!

Reply:
> As you can see here, there's an error in the code that pretty prints the class, it prints the `class <class_name>` header twice!

Hello @NathanReb, i have challenges figuring this out. Simply deleting class add_class from the code you pointed out above does not help. i tried filtering with the code below 
  $ api-diff ref_class.cmi remove_class.cmi | sed 's/class [a-z_]\+: class /class /'
  diff module Remove_class:
  -type ref_class = < get : int; set : int -> unit >
  -class ref_class: object method get : int method set : int -> unit end, which results in errors. Please i would need your assistance here.

Reply:
> Simply deleting class add_class from the code you pointed out above does not help.

What do you mean with "the code you point out above"?

The code on which the github review comment is attached is the test part. What @NathanReb highlight, is that the output of the test is not what we want: we do not want the duplicated "`class`" in the output.
This is why we test things: we check that the output is the expected one. If it's not what we expect, then there is a bug to fix. But the solution you propose is to change the test, to cancel the bug with string manipulation!

So, the test is right, what you should fix is the bug that the test highlights. Find why there is a duplicated `class`, fix the bug, then run `dune test --auto-promote` to update the test to the new output, after the bugfix.

Reply:
OK, thank you.

Reply:
Hello @NathanReb, @panglesd. I figured this out. The issue resulted from how cd-to-line was defined in 

> > Simply deleting class add_class from the code you pointed out above does not help.
> 
> What do you mean with "the code you point out above"?
> 
> The code on which the github review comment is attached is the test part. What @NathanReb highlight, is that the output of the test is not what we want: we do not want the duplicated "`class`" in the output. This is why we test things: we check that the output is the expected one. If it's not what we expect, then there is a bug to fix. But the solution you propose is to change the test, to cancel the bug with string manipulation!
> 
> So, the test is right, what you should fix is the bug that the test highlights. Find why there is a duplicated `class`, fix the bug, then run `dune test --auto-promote` to update the test to the new output, after the bugfix.

text_diff.ml. Thank you so much for the support. I currently working handling the merge conflicts.